### PR TITLE
Improve robot window documentation

### LINKED ISSUE #1055: Improve HTML robot-window Doc
**Documentation page(s) concerned**
https://www.cyberbotics.com/doc/guide/controller-plugin#robot-window

**What is the problem with the current documentation?**
The description on this page is way not sufficient, we should write a small how-to or improve this doc to help users creating their own robot-window.
Furthermore, instead of pointing to the thymio robot window we should point to the 'custom_robot_window' sample one.

### PR DESCRIPTION
**Pull request by lukicdarkoo**
Fixes #1055

**Documentation**
- https://cyberbotics.com/doc/guide/samples-howto?version=feature-robot-window-doc#custom_robot_window_simple-wbt
- https://cyberbotics.com/doc/guide/controller-plugin?version=feature-robot-window-doc#robot-window

**TODO**
- [x] Add a very basic robot window
- [x] Refactor the `projects/samples/howto` directory, so each example is in a dedicated folder
  - [x] Change the documentation accordingly
- [x] Delete extra Makefiles (e.g. `projects/samples/howto/plugins/physics/Makefile`) to minimize the noise
- [x] Fix the `custom_html_robot_window` example (it wasn't working)
- [x] Create a tutorial
- [x] Add to changelog
- [x] Verify regressions

### CODE REVIEW

**Comment by stefaniapedrazzi** [comment]
> Delete the force_control_matlab controller (not referenced anywhere)
    Please let me know if it should be reverted

If I'm not wrong this was added to provide some controller samples in MATLAB, in fact it is automatically open in the editor when loading the world.
It cannot be set as default one because otherwise users without MATLAB cannot use it.
Instead of removing it, it would probably be better to create a new world with `_matlab` suffix that use it by default as we did for other worlds.

**Comment by lukicdarkoo** [comment]
> > Delete the force_control_matlab controller (not referenced anywhere)
> > Please let me know if it should be reverted
> 
> If I'm not wrong this was added to provide some controller samples in MATLAB, in fact it is automatically open in the editor when loading the world.
> It cannot be set as default one because otherwise users without MATLAB cannot use it.
> Instead of removing it, it would probably be better to create a new world with `_matlab` suffix that use it by default as we did for other worlds.

To avoid world duplication I can mention in `samples-howto.md` that the MATLAB controller can be chosen instead?

File: docs/guide/controller-plugin.md
Comment (omichel) [inline docs/guide/controller-plugin.md:71]:
```suggestion
A simple example of an HTML robot window is given in [`projects/samples/howto/custom_robot_window_simple`](samples-howto.md#custom_robot_window_simple-wbt) and it demonstrates how to establish a two-way communication between a robot window and Python controller.
```

File: docs/guide/controller-plugin.md
Comment (omichel) [inline docs/guide/controller-plugin.md:76]: I would put a hyperlink to the github file here (like is done afterwards).

File: docs/guide/controller-plugin.md
Comment (omichel) [inline docs/guide/controller-plugin.md:81]: I would put a link here as well.

File: docs/guide/samples-howto.md
Comment (omichel) [inline docs/guide/samples-howto.md:51]:
```suggestion
![custom_robot_window_simple.png](images/samples/custom_robot_window_simple.thumbnail.jpg) This example shows a minimal two way communication between a simple robot window and a robot controller.
```

**Review by omichel — CHANGES_REQUESTED** [review]
Thank you.
It looks overall very good to me.
Note: I fear we will have many conflicts when merging the released branch into the master branch (because of the released URLs). Maybe, we should do it quickly after (or before) merging this PR.

**Comment by lukicdarkoo** [comment]
> Thank you.
> It looks overall very good to me.
> Note: I fear we will have many conflicts when merging the released branch into the master branch (because of the released URLs). Maybe, we should do it quickly after (or before) merging this PR.

Let's wait for @stefaniapedrazzi review and then we can sync the branches quickly.

**Review by omichel — APPROVED** [review]

**Review by stefaniapedrazzi — CHANGES_REQUESTED** [review]
The following file has to be updated. The compilation of the projects folder is currently failing on my system.
https://github.com/cyberbotics/webots/blob/feature-robot-window-doc/projects/samples/Makefile

I also started the tests on this branch.
They should report the same issue too.

File: docs/guide/controller-plugin.md
Comment (stefaniapedrazzi) [inline docs/guide/controller-plugin.md:76]: This is correct in this PR.
But note that since the introduction of the `released` branch we have now issues with this kind of links because we need to check and update (if new have been added) every time we release a new version.

File: docs/guide/controller-plugin.md
Comment (stefaniapedrazzi) [inline docs/guide/controller-plugin.md:83]:
```suggestion
An example of a reusable HTML robot window is given in [`projects/samples/howto/custom_robot_window`](samples-howto.md#custom_robot_window-wbt).
```

File: docs/guide/controller-plugin.md
Comment (lukicdarkoo) [reply docs/guide/controller-plugin.md:76]: But this problem is not specific to this PR, it is project-wide, this is not the only occurrence of a GitHub link. Maybe, we can improve the documentation engine to handle branches automatically? Like if we write:
```md
[`path/to/file`](https://github.com/cyberbotics/webots/tree/{version}/path/to/file)
```
it should resolve `{version}` placeholder as a branch name.

File: docs/guide/controller-plugin.md
Comment (stefaniapedrazzi) [inline docs/guide/controller-plugin.md:84]:
```suggestion
While in the previous example the HTML robot window interacts directly with the controller, in this example the robot window exchanges data with a controller implemented in [`plugins/robot_windows/custom_robot_window/custom_robot_window.c`](https://github.com/cyberbotics/webots/tree/master/projects/samples/howto/custom_robot_window/plugins/robot_windows/custom_robot_window/custom_robot_window.c).
```

File: docs/guide/controller-plugin.md
Comment (stefaniapedrazzi) [inline docs/guide/controller-plugin.md:85]: In the resulting documentation page, there is no empty line between this sentence and the next one.

File: docs/guide/controller-plugin.md
Comment (stefaniapedrazzi) [reply docs/guide/controller-plugin.md:76]: Yes, it is not specific to this PR but this is the first time we are re-introducing this issue.
I just wanted to point out the problem because after merging this we should fix it again.
We should open an issue and fix this as soon as possible.

**Review by stefaniapedrazzi — CHANGES_REQUESTED** [review]
There are still some path to fix: 
* https://github.com/cyberbotics/webots/blob/feature-robot-window-doc/docs/reference/contactproperties.md
* https://github.com/cyberbotics/webots/blob/feature-robot-window-doc/docs/reference/motor.md
* https://github.com/cyberbotics/webots/blob/feature-robot-window-doc/docs/reference/supervisor.md
* https://github.com/cyberbotics/webots/blob/feature-robot-window-doc/scripts/converter/convert_all_worlds.py

File: docs/guide/controller-plugin.md
Comment (stefaniapedrazzi) [reply docs/guide/controller-plugin.md:76]: Yes, it is not specific to this PR. But this is the first time we introduce again a link that we will need to fix later.
I just wanted to point out the problem and warn that this will need to be fixed after merging it.
It would be good to open an issue about this and fix it as soon as possible.

**Comment by lukicdarkoo** [comment]
Btw, `projects/samples/howto/supervisor/worlds/supervisor.wbt` seems to be deleted before:
https://github.com/cyberbotics/webots/tree/master/projects/samples/howto

File: docs/guide/controller-plugin.md
Comment (lukicdarkoo) [reply docs/guide/controller-plugin.md:85]: image
It looks good to me. Do we refer to the same thing?

Comment (stefaniapedrazzi) [reply docs/guide/controller-plugin.md:85]: I would have expected that the `The HTML robot windows..` part would be in a different paragraph so that it is clear that is applies to both the robot windows.

Comment (lukicdarkoo) [reply docs/guide/controller-plugin.md:85]: I added an empty line explicitly.

However, those were two different paragraphs according to markdown. GitHub renders them as two different paragraphs:
image

**Comment by stefaniapedrazzi** [comment]
I forgot that we should also add an entry in the changelog about the new structure of the "howto" folder.

**Review by stefaniapedrazzi — APPROVED** [review]
Thank you.

Could you please open an issue for the link containing the branch name?

**Pull request merged**
